src/reporter/JestReporter.ts

```diff
@@ -0,0 +1,247 @@
+        this.aggregatedResults.numTotalTestSuites = numSuites;
+
+        this.reporters.forEach((reporter) => {
+            reporter.onRunStart(this.aggregatedResults, {
+                estimatedTime: 0, // TODO: estimate this maybe?
+                showStatus: false,
+            });
+        });
+    }
+
+    /**
+     * Callback for when all test files have been executed.
+     */
+    public onRunComplete() {
+        let contextSet = new Set([createContext(this.projectConfig)]);
+        this.reporters.forEach((reporter) => {
+            reporter.onRunComplete(contextSet, this.aggregatedResults);
+        });
+    }
+
+    /**
+     * Callback for when brs encounters an execution error.
+     * @param filename The name of the file that failed
+     * @param index The TAP index of the file
+     * @param reason The error that was thrown to cause the execution error
+     */
+    public onFileExecError(
+        filename: string,
+        index: number,
+        reason: Error | BrsError | BrsError[]
+    ) {
+        // If we get an array of errors, report the first one.
+        reason = Array.isArray(reason) ? reason[0] : reason;
+
+        if (isBrsError(reason)) {
+            this.currentResults.testExecError = {
+                stack: `\nat ${reason.location.file}:${reason.location.start.line}:${reason.location.start.column}`,
+                message: reason.message,
+            };
+        } else {
+            this.currentResults.testExecError = {
+                stack: reason.stack,
+                message: reason.message,
+            };
+        }
+    }
+
+    /**
+     * Callback for when a file starts test execution. Creates empty results
+     * for the new file and sets it as the current results for this parser.
+     * @param filePath The path to the file that is starting its run.
+     */
+    public onFileStart(filePath: string) {
+        this.currentResults = createEmptyTestResult();
+        this.currentResults.testFilePath = filePath;
+        this.currentResults.perfStats.start = Date.now();
+    }
+
+    /**
+     * Callback for when a file ends execution. Adds the results from the current
+     * file to the aggregated results, and tells each Jest reporter that this file has completed.
+     */
+    public onFileComplete() {
+        // Flatten console output because tap-parser splits output by newline in the "extra" event.
+        // We don't know which lines are supposed to go together, so just print them all together.
+        if (this.currentResults.console) {
+            this.currentResults.console = [
+                {
+                    message: this.currentResults.console
+                        .map((entry) => entry.message)
+                        .join(""),
+                    origin: "", // TODO: figure out how to get the stack trace
+                    type: "print" as any, // a little hack because brightscript uses `print` instead of `console.log`
```

Comment:
Aw bummer it reports as `console.print` instead of just `print`?  So close!

Reply:
I know, so close 😢 And we have no control over the outputted string here, either, so we can't use `replace` or anything, sadly.